src/configuration/remapper.ts

```diff
@@ -56,17 +56,49 @@ export class Remappers implements IRemapper {
 export class Remapper implements IRemapper {
   private readonly _configKey: string;
   private readonly _remappedModes: Mode[];
-  private readonly _recursive: boolean;
   private readonly _logger = Logger.get('Remapper');
 
+  /**
+   * Checks if the current commandList is a potential remap.
+   */
   private _isPotentialRemap = false;
+
+  /**
+   * If the commandList has a remap but there is still another potential remap we
+   * call it an Ambiguous Remap and we store it here. If later we need to handle it
+   * we don't need to go looking for it.
+   */
+  private _hasAmbiguousRemap: false | IKeyRemapping = false;
```

Comment:
Any reason this isn't `IKeyRemapping | undefined`?

Reply:
Not really, I just thought it was easier to understand as it was an answer to the variable name `_hasAmbiguousRemap? false` but I can change it to undefined if you think it's better.

Reply:
It also is more on par with this other variable
```ts
private _hasPotentialRemap = false;
```
I think it looks better to do something like this:
```ts
this._hasAmbiguousRemap = false;
this._hasPotentialRemap = false;
```
than this:
```ts
this._hasAmbiguousRemap = undefined;
this._hasPotentialRemap = false;
```

But if you still think it's better to go with undefined tell me and I'll change it.

Reply:
Personally I'd write it as `_ambiguousRemap: IKeyRemapping | undefined` and check if we have one by checking for `undefined`, but I'll leave it up to you

Reply:
I did change it to `undefined` but I kept it as `_hasAmbiguousRemap` because I think it makes it easier to read and understand in some of those confusing if statements.